tests/quantization/torchao/test_torchao.py

```diff
@@ -404,43 +414,57 @@ def test_training(self):
-        transformer_int4wo_gs32 = self.get_dummy_components(TorchAoConfig("int4wo", group_size=32))["transformer"]
-        transformer_int8wo = self.get_dummy_components(TorchAoConfig("int8wo"))["transformer"]
-        transformer_bf16 = self.get_dummy_components(None)["transformer"]
-
-        total_int4wo = get_model_size_in_bytes(transformer_int4wo)
-        total_int4wo_gs32 = get_model_size_in_bytes(transformer_int4wo_gs32)
-        total_int8wo = get_model_size_in_bytes(transformer_int8wo)
-        total_bf16 = get_model_size_in_bytes(transformer_bf16)
-
-        # Latter has smaller group size, so more groups -> more scales and zero points
-        self.assertTrue(total_int4wo < total_int4wo_gs32)
-        # int8 quantizes more layers compare to int4 with default group size
-        self.assertTrue(total_int8wo < total_int4wo)
-        # int4wo does not quantize too many layers because of default group size, but for the layers it does
-        # there is additional overhead of scales and zero points
-        self.assertTrue(total_bf16 < total_int4wo)
+        for model_id in ["hf-internal-testing/tiny-flux-pipe", "hf-internal-testing/tiny-flux-sharded"]:
+            transformer_int4wo = self.get_dummy_components(TorchAoConfig("int4wo"), model_id=model_id)["transformer"]
+            transformer_int4wo_gs32 = self.get_dummy_components(
+                TorchAoConfig("int4wo", group_size=32), model_id=model_id
+            )["transformer"]
+            transformer_int8wo = self.get_dummy_components(TorchAoConfig("int8wo"), model_id=model_id)["transformer"]
+            transformer_bf16 = self.get_dummy_components(None, model_id=model_id)["transformer"]
+
+            self.assertTrue(
+                isinstance(transformer_int4wo.transformer_blocks[0].ff.net[2].weight, AffineQuantizedTensor)
+            )
+            self.assertTrue(
+                isinstance(transformer_int4wo_gs32.transformer_blocks[0].ff.net[2].weight, AffineQuantizedTensor)
+            )
+            self.assertTrue(
+                isinstance(transformer_int8wo.transformer_blocks[0].ff.net[2].weight, AffineQuantizedTensor)
+            )
+
+            total_int4wo = get_model_size_in_bytes(transformer_int4wo)
+            total_int4wo_gs32 = get_model_size_in_bytes(transformer_int4wo_gs32)
+            total_int8wo = get_model_size_in_bytes(transformer_int8wo)
+            total_bf16 = get_model_size_in_bytes(transformer_bf16)
+
+            # Latter has smaller group size, so more groups -> more scales and zero points
+            self.assertTrue(total_int4wo < total_int4wo_gs32)
+            # int8 quantizes more layers compare to int4 with default group size
+            self.assertTrue(total_int8wo < total_int4wo)
+            # int4wo does not quantize too many layers because of default group size, but for the layers it does
+            # there is additional overhead of scales and zero points
+            self.assertTrue(total_bf16 < total_int4wo)
```

Comment:
Possible to do like this?

https://github.com/huggingface/diffusers/blob/825979ddc3d03462287f1f5439e89ccac8cc71e9/tests/quantization/bnb/test_4bit.py#L175-L180

Helps make the tests a bit more uniform across different backends.

Reply:
Could do it in a follow-up PR because it current works as expected at the moment and don't want to introduce any new test-related bugs experimenting with an alternative implementation here

Reply:
Okay for me to do that in a follow-up. I can also look into it. Could you at least add a TODO then?  

I think we still don't have an upper bound on the expected memory savings here, no? Like `total_bf16 < total_int4wo` would probably be always True but how much is the actual memory saving a user should expect here? Is it not that deterministic in `torchao`? 

I do think it's related to the fix we're trying to ship here as it lets us know upfront what to expect.

Reply:
Note that this is a dummy model and there are no memory savings in this case (bf16 takes lower memory than int4wo) because the shapes of the linear weights are not compatible.

In the ideal case, for almost any real model that we try, int4 requires quarter the memory of the bf16 weights. The test only serves to better understand that the quantization method adds other tensors apart from the weights itself. For dummy models like this one, the total size therefore exceeds even the normal bf16 weights with no quantization.

> Is it not that deterministic in torchao?

For most cases, and on real models that are nicely aligned with layer sizes of `2^N`, it is deterministic. Some quantization methods have additional constraints required at the moment - one such example being int4wo. If you try quantizing a `(1000, 1000)` linear with int4wo group_size=64, it won't get quantized and there will be no memory benefits. If you tried, however, to quantize a `(1024, 1024)` linear, it should take about 1/8th the memory of fp32 weights because now the dimensions are divisible by 64.

Reply:
This is good info. Perhaps in the follow-up PR, we could add a memory footprint test on a real model like we do for `bitsandbytes`. 

I do think that test deserves to be here or are we already testing it? Would like to take opinion of @DN6 here too.

Reply:
The slow tests for Flux do check expected memory for both int4wo (should be 1/4 of bf16 weights) and int8wo (should be 1/2 of bf16 weights). Do you have something else in mind?

Reply:
Ah perfect! That should work, thanks Aryan!